presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastoreConfig.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.metastore.thrift;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.Duration;
+import io.prestosql.plugin.hive.RetryDriver;
+
+import javax.validation.constraints.NotNull;
+
+public class ThriftHiveMetastoreConfig
+{
+    private int maxRetries = RetryDriver.DEFAULT_RETRY_ATTEMPTS - 1;
+    private double backoffScaleFactor = RetryDriver.DEFAULT_SCALE_FACTOR;
+    private Duration minBackoffDelay = RetryDriver.DEFAULT_SLEEP_TIME;
+    private Duration maxBackoffDelay = RetryDriver.DEFAULT_SLEEP_TIME;
+    private Duration maxRetryTime = RetryDriver.DEFAULT_MAX_RETRY_TIME;
+
+    @NotNull
+    public int getMaxRetries()
+    {
+        return maxRetries;
+    }
+
+    @Config("hive.metastore.thrift.client.max-retries")
+    @ConfigDescription("Maximum number of retry attempts for metastore requests")
+    public ThriftHiveMetastoreConfig setMaxRetries(int maxRetries)
+    {
+        this.maxRetries = maxRetries;
+        return this;
+    }
+
+    @NotNull
```

Comment:
No need for `@NotNull` since this is a primitive value.

Reply:
OK